### PR TITLE
Issue #13100: Create tests from design category checks xdoc examples

### PR DESCRIPTION
**Pull request by stoyanK7**
Part of #13100 

https://checkstyle.org/checks/design/index.html

### CODE REVIEW

**Comment by stoyanK7** [comment]
Moving to draft, I will do this for the whole design category.

File: src/xdocs/checks/design/onetoplevelclass.xml.template [hunk 1]
```diff
@@ -20,58 +20,42 @@
...
-        <p>
+        <macro name="example">
+          <param name="path"
+                 value="resources/com/puppycrawl/tools/checkstyle/checks/design/onetoplevelclass/Example1.txt"/>
+          <param name="type" value="code"/>
+        </macro>
+        <p id="Example2-code">
           An example of code without public top-level type:
         </p>
-        <div class="wrapper">
```
Comment (stoyanK7) [inline src/xdocs/checks/design/onetoplevelclass.xml.template:54]: These `wrapper` divs are only used for tables as far as I am aware. I assume this was the old way of showing code in the website and some examples didn't get converted to `<source>`

**Review by romani — APPROVED** [review]
Ok to merge

**Comment by nrmancuso** [comment]
Github, generate site

File: src/xdocs/checks/design/onetoplevelclass.xml.template (hunk 1 above)
Comment (nrmancuso) [reply src/xdocs/checks/design/onetoplevelclass.xml.template:54]: I think using `wrapper` has a bad side effect too, compare these two examples.

From the currently published website (https://checkstyle.org/checks/design/onetoplevelclass.html) :
image

From the site generated in this PR (https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/a63cc64_2023095803/checks/design/onetoplevelclass.html#Examples) :
image

It seems like wrapper does not enclose the code examples in the "frame" like `source` does. @stoyanK7 thanks for finding this.

**Review by nrmancuso — APPROVED** [review]

File: src/xdocs/checks/design/onetoplevelclass.xml.template (hunk 1 above)
Comment (nrmancuso) [reply src/xdocs/checks/design/onetoplevelclass.xml.template:54]: @romani ^

**Pull request merged**